CMakeLists.txt

```diff
@@ -12,16 +12,23 @@ set(GCF_IGNORE_LIST "" CACHE STRING
     "Semi colon separated list of directories to ignore")
 set(GCF_CLANGFORMAT_MINIMAL_VERSION "0.0.0" CACHE STRING
     "The minimal required version of clang-format")
-set(GCF_FORCE_OVERWRITE OFF CACHE BOOL
-    "If true, always overwrite pre-commit hook and script")
+if(NOT GCF_FORCE_OVERWRITE)
+  set(GCF_FORCE_OVERWRITE OFF CACHE BOOL
+      "If true, always overwrite pre-commit hook and script")
+endif()
 set(GCF_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/git-cmake-format.py)
+if(NOT GCF_GIT_TOP_LEVEL_DIR)
+  message(FATAL
+          "Need to specify the top-level directory of the git repository "
+          "into which the hooks should be installed.")
+endif()
 
 find_package(ClangFormat ${GCF_CLANGFORMAT_MINIMAL_VERSION} REQUIRED)
 
 execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --git-path hooks
-  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-  OUTPUT_VARIABLE GCF_GIT_HOOKS_DIR
-  OUTPUT_STRIP_TRAILING_WHITESPACE)
+                WORKING_DIRECTORY ${GCF_TOP_LEVEL_DIR}
```

Comment:
why not `PROJECT_SOURCE_DIR`?

Reply:
Because in our workflow we use gcf through FetchContent, so that will point to the build directory of the top-most project. But we need some way to figure out that gcf was called from ginkgo and thus should only modify the hooks for ginkgo. I didn't find out how to get details on the cmake project hierarchy that you are currently in, so I just force the user to set that variable.